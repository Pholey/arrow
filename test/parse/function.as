def method() {
}

def method(): int64 {
}

// def method(): (int64, int64) -> int64 {
// }

def method(a: int64, b: int64) {
}

def method(a: int64, b: int64, c: int64 = 310) {
}

def method(a: int64, b: int64, c: int64 = 310): bool {
}
