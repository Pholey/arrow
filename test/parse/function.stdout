{
  "tag": "Module",
  "span": "test/parse/function.as:1-30",
  "statements": [
    {
      "tag": "Function",
      "span": "test/parse/function.as:4-5",
      "name": "methodA",
      "result_type": null,
      "parameters": [],
      "type_parameters": [],
      "block": {
        "tag": "Block",
        "span": "test/parse/function.as:4-5",
        "statements": []
      }
    },
    {
      "tag": "Function",
      "span": "test/parse/function.as:7-9",
      "name": "methodB",
      "result_type": {
        "tag": "TypeName",
        "span": "test/parse/function.as:7:16-21",
        "name": {
          "tag": "Identifier",
          "span": "test/parse/function.as:7:16-21",
          "text": "int64"
        }
      },
      "parameters": [],
      "type_parameters": [],
      "block": {
        "tag": "Block",
        "span": "test/parse/function.as:7-9",
        "statements": [
          {
            "tag": "Return",
            "span": "test/parse/function.as:8:3-12",
            "operand": {
              "tag": "Integer",
              "span": "test/parse/function.as:8:10-12",
              "value": "40"
            }
          }
        ]
      }
    },
    {
      "tag": "Function",
      "span": "test/parse/function.as:11-13",
      "name": "methodC",
      "result_type": {
        "tag": "TypeFunction",
        "span": "test/parse/function.as:11:16-39",
        "parameters": [
          {
            "tag": "TypeFunctionParameter",
            "span": "test/parse/function.as:11:17-22",
            "type": {
              "tag": "TypeName",
              "span": "test/parse/function.as:11:17-22",
              "name": {
                "tag": "Identifier",
                "span": "test/parse/function.as:11:17-22",
                "text": "int64"
              }
            },
            "is_mutable": false
          },
          {
            "tag": "TypeFunctionParameter",
            "span": "test/parse/function.as:11:24-29",
            "type": {
              "tag": "TypeName",
              "span": "test/parse/function.as:11:24-29",
              "name": {
                "tag": "Identifier",
                "span": "test/parse/function.as:11:24-29",
                "text": "int64"
              }
            },
            "is_mutable": false
          }
        ],
        "result": {
          "tag": "TypeName",
          "span": "test/parse/function.as:11:34-39",
          "name": {
            "tag": "Identifier",
            "span": "test/parse/function.as:11:34-39",
            "text": "int64"
          }
        }
      },
      "parameters": [],
      "type_parameters": [],
      "block": {
        "tag": "Block",
        "span": "test/parse/function.as:11-13",
        "statements": [
          {
            "tag": "Return",
            "span": "test/parse/function.as:12:3-17",
            "operand": {
              "tag": "Identifier",
              "span": "test/parse/function.as:12:10-17",
              "text": "methodD"
            }
          }
        ]
      }
    },
    {
      "tag": "Function",
      "span": "test/parse/function.as:15-17",
      "name": "methodD",
      "result_type": {
        "tag": "TypeName",
        "span": "test/parse/function.as:15:34-39",
        "name": {
          "tag": "Identifier",
          "span": "test/parse/function.as:15:34-39",
          "text": "int64"
        }
      },
      "parameters": [
        {
          "tag": "Parameter",
          "span": "test/parse/function.as:15:13-21",
          "name": "a",
          "type": {
            "tag": "TypeFunctionParameter",
            "span": "test/parse/function.as:15:16-21",
            "type": {
              "tag": "TypeName",
              "span": "test/parse/function.as:15:16-21",
              "name": {
                "tag": "Identifier",
                "span": "test/parse/function.as:15:16-21",
                "text": "int64"
              }
            },
            "is_mutable": false
          }
        },
        {
          "tag": "Parameter",
          "span": "test/parse/function.as:15:23-31",
          "name": "b",
          "type": {
            "tag": "TypeFunctionParameter",
            "span": "test/parse/function.as:15:26-31",
            "type": {
              "tag": "TypeName",
              "span": "test/parse/function.as:15:26-31",
              "name": {
                "tag": "Identifier",
                "span": "test/parse/function.as:15:26-31",
                "text": "int64"
              }
            },
            "is_mutable": false
          }
        }
      ],
      "type_parameters": [],
      "block": {
        "tag": "Block",
        "span": "test/parse/function.as:15-17",
        "statements": [
          {
            "tag": "Return",
            "span": "test/parse/function.as:16:3-19",
            "operand": {
              "tag": "Call",
              "span": "test/parse/function.as:16:17-19",
              "operand": {
                "tag": "Identifier",
                "span": "test/parse/function.as:16:10-17",
                "text": "methodB"
              },
              "arguments": []
            }
          }
        ]
      }
    },
    {
      "tag": "Variable",
      "span": "test/parse/function.as:22:1-17",
      "name": "a",
      "mutable": false,
      "type": {
        "tag": "TypeFunction",
        "span": "test/parse/function.as:22:8-16",
        "parameters": [],
        "result": {
          "tag": "TypeUnit",
          "span": "test/parse/function.as:22:14-16"
        }
      },
      "initializer": null
    },
    {
      "tag": "Variable",
      "span": "test/parse/function.as:25:1-31",
      "name": "b",
      "mutable": false,
      "type": {
        "tag": "TypeFunction",
        "span": "test/parse/function.as:25:8-30",
        "parameters": [
          {
            "tag": "TypeFunctionParameter",
            "span": "test/parse/function.as:25:9-14",
            "type": {
              "tag": "TypeName",
              "span": "test/parse/function.as:25:9-14",
              "name": {
                "tag": "Identifier",
                "span": "test/parse/function.as:25:9-14",
                "text": "int32"
              }
            },
            "is_mutable": false
          },
          {
            "tag": "TypeFunctionParameter",
            "span": "test/parse/function.as:25:16-21",
            "type": {
              "tag": "TypeName",
              "span": "test/parse/function.as:25:16-21",
              "name": {
                "tag": "Identifier",
                "span": "test/parse/function.as:25:16-21",
                "text": "int64"
              }
            },
            "is_mutable": false
          }
        ],
        "result": {
          "tag": "TypeName",
          "span": "test/parse/function.as:25:26-30",
          "name": {
            "tag": "Identifier",
            "span": "test/parse/function.as:25:26-30",
            "text": "bool"
          }
        }
      },
      "initializer": null
    },
    {
      "tag": "Variable",
      "span": "test/parse/function.as:30:1-56",
      "name": "c",
      "mutable": false,
      "type": {
        "tag": "TypeFunction",
        "span": "test/parse/function.as:30:8-55",
        "parameters": [
          {
            "tag": "TypeFunctionParameter",
            "span": "test/parse/function.as:30:9-13",
            "type": {
              "tag": "TypeName",
              "span": "test/parse/function.as:30:9-13",
              "name": {
                "tag": "Identifier",
                "span": "test/parse/function.as:30:9-13",
                "text": "bool"
              }
            },
            "is_mutable": false
          }
        ],
        "result": {
          "tag": "TypeFunction",
          "span": "test/parse/function.as:30:18-55",
          "parameters": [
            {
              "tag": "TypeFunctionParameter",
              "span": "test/parse/function.as:30:19-24",
              "type": {
                "tag": "TypeName",
                "span": "test/parse/function.as:30:19-24",
                "name": {
                  "tag": "Identifier",
                  "span": "test/parse/function.as:30:19-24",
                  "text": "int32"
                }
              },
              "is_mutable": false
            }
          ],
          "result": {
            "tag": "TypeFunction",
            "span": "test/parse/function.as:30:29-55",
            "parameters": [
              {
                "tag": "TypeFunctionParameter",
                "span": "test/parse/function.as:30:30-45",
                "type": {
                  "tag": "TypeFunction",
                  "span": "test/parse/function.as:30:30-45",
                  "parameters": [
                    {
                      "tag": "TypeFunctionParameter",
                      "span": "test/parse/function.as:30:31-35",
                      "type": {
                        "tag": "TypeName",
                        "span": "test/parse/function.as:30:31-35",
                        "name": {
                          "tag": "Identifier",
                          "span": "test/parse/function.as:30:31-35",
                          "text": "bool"
                        }
                      },
                      "is_mutable": false
                    }
                  ],
                  "result": {
                    "tag": "TypeName",
                    "span": "test/parse/function.as:30:40-45",
                    "name": {
                      "tag": "Identifier",
                      "span": "test/parse/function.as:30:40-45",
                      "text": "int32"
                    }
                  }
                },
                "is_mutable": false
              }
            ],
            "result": {
              "tag": "TypeName",
              "span": "test/parse/function.as:30:50-55",
              "name": {
                "tag": "Identifier",
                "span": "test/parse/function.as:30:50-55",
                "text": "int32"
              }
            }
          }
        }
      },
      "initializer": null
    }
  ]
}
